pandas/core/indexes/datetimelike.py

```diff
@@ -732,8 +737,15 @@ def __rsub__(self, other):
             return -(self - other)
         cls.__rsub__ = __rsub__
 
-        cls.__iadd__ = __add__
-        cls.__isub__ = __sub__
+        def __iadd__(self, other):
+            # alias for __add__
+            return self.__add__(other)
+        cls.__iadd__ = __iadd__
```

Comment:
why do we need this double assignment ?

Reply:
If we just set `cls.__iadd__ = __add__` then when we check for `Index.__iadd__.__name__` we'll get `__add__` instead of `__iadd__`.  Not a big deal, but its cheap to make it pretty.